pkg/monitor/backenddisruption/disruption_backend_sampler.go

```diff
@@ -198,6 +198,10 @@ func (b *BackendSampler) bodyMatches(body []byte) error {
 	return nil
 }
 
+func (b *BackendSampler) SetHost(host string) {
+	b.host = host
+}
```

Comment:
Is breaking the "With" pattern above intentional due to the late binding nature of it?

Reply:
> Is breaking the "With" pattern above intentional due to the late binding nature of it?

yes